doi-requests-commons/src/main/java/no/unit/nva/doi/requests/handlers/DoiRequestAuthorizedHandlerTemplate.java

```diff
@@ -0,0 +1,78 @@
+package no.unit.nva.doi.requests.handlers;
+
+import static nva.commons.utils.attempt.Try.attempt;
+import com.amazonaws.services.securitytoken.AWSSecurityTokenService;
+import com.amazonaws.services.securitytoken.AWSSecurityTokenServiceClientBuilder;
+import com.amazonaws.services.securitytoken.model.Tag;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Locale;
+import java.util.stream.Collectors;
+import no.unit.nva.doi.requests.service.impl.DynamoDbDoiRequestsServiceFactory;
+import nva.commons.handlers.AuthorizedApiGatewayHandler;
+import nva.commons.handlers.RequestInfo;
+import nva.commons.utils.Environment;
+import nva.commons.utils.JacocoGenerated;
+import org.slf4j.Logger;
+
+public abstract class DoiRequestAuthorizedHandlerTemplate<I, O> extends AuthorizedApiGatewayHandler<I, O> {
+
+    public static final String PUBLISHER_IDENTIFIER = "publisherIdentifier";
+    protected static final DynamoDbDoiRequestsServiceFactory DEFAULT_SERVICE_FACTORY = defaultServiceFactory();
+
+    @JacocoGenerated
+    protected DoiRequestAuthorizedHandlerTemplate(Class<I> iclass, Environment environment,
+                                                  AWSSecurityTokenService stsClient,
+                                                  Logger logger) {
+        super(iclass, environment, stsClient, logger);
+    }
+
+    @Override
+    protected final List<Tag> sessionTags(RequestInfo requestInfo) {
+
+        List<Tag> accessRightsTags = accessRightsToTags(requestInfo);
```

Comment:
a tag of an access right will have the form `"approve_doi_request": "APPROVE_DOI_REQUEST"`. This will be transformed in the following rule:
```
StringEqualsIgnoreCase:
                    "aws:PrincipalTag/approve_doi_request": "APPROVE_DOI_REQUEST"
```

 We only care if the access right is present so this maybe a convenient way to ensure that the tag key is not arbitrary and is determined by the access right itself.